[quant][eagermode] Avoid inserting fakequant for sigmoid/hardsigmoid/tanh in eval mode

jerryzh168 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#47297 [quant][eagermode] Avoid inserting fakequant for sigmoid/hardsigmoid/tanh in eval mode**

Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D24708270](https://our.internmc.facebook.com/intern/diff/D24708270)

jerryzh168 committed:
[quant][eagermode] Avoid inserting fakequant for sigmoid/hardsigmoid/tanh in eval mode

Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

z-a-f commented on torch/quantization/quantization_mappings.py:
```diff
@@ -179,5 +179,5 @@ def get_default_special_act_post_process(module_cls):
 
-def has_special_act_post_process(module_cls):
-    return module_cls in DEFAULT_MODULE_TO_ACT_POST_PROCESS
+def has_special_act_post_process(module):
+    return module.training and type(module) in DEFAULT_MODULE_TO_ACT_POST_PROCESS
```
nit :)

I feel like in cases like this we should check if it is a class or an instance. You can do it using `inspect.isclass`. Here is the behavior:

```python
import inspect
class Foo:
  pass
f = Foo()
inspect.isclass(f)  # Returns "False"
inspect.isclass(Foo)  # Returns "True"
``` 

That way in case someone tries to pass a type, it will be handled

```python
if inspect.isclass(module):
  raise ValueError(f"Expecting an instance of a class, got {module}")
```

jerryzh168 replied:
this is probably too complicated for a small helper function :) but thanks for the suggestion. I'll change this to be an internal helper function.

z-a-f reviewed (approved):
LGTM

jerryzh168 committed:
Update on "[quant][eagermode] Avoid inserting fakequant for sigmoid/hardsigmoid/tanh in eval mode"


Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D24708270](https://our.internmc.facebook.com/intern/diff/D24708270)

[ghstack-poisoned]
